src/test/javascript/spec/component/exam/manage/exams/exam-detail.component.spec.ts

```diff
@@ -0,0 +1,120 @@
+                    { path: 'course-management/:courseId/exams/:examId/student-exams', component: DummyComponent },
+                ]),
+            ],
+            declarations: [
+                ExamDetailComponent,
+                DummyComponent,
+                MockPipe(TranslatePipe),
+                MockPipe(ArtemisDatePipe),
+                MockComponent(AlertComponent),
+                MockComponent(AlertErrorComponent),
+                MockComponent(FaIconComponent),
+                MockDirective(JhiTranslateDirective),
+            ],
+            providers: [
+                {
+                    provide: ActivatedRoute,
+                    useValue: {
+                        data: {
+                            subscribe: (fn: (value: Data) => void) =>
+                                fn({
+                                    exam,
+                                }),
+                        },
+                    },
+                },
+                MockProvider(AccountService, {
+                    isAtLeastInstructorInCourse: () => true,
+                }),
+                MockProvider(ArtemisMarkdownService, {
+                    safeHtmlForMarkdown: () => exampleHTML,
+                }),
+            ],
+            schemas: [],
+        })
+            .compileComponents()
+            .then(() => {
+                examDetailComponentFixture = TestBed.createComponent(ExamDetailComponent);
+                examDetailComponent = examDetailComponentFixture.componentInstance;
+            });
+    });
+
+    afterEach(function () {
+        sinon.restore();
+    });
+
+    it('should load exam from route and display it to user', () => {
+        examDetailComponentFixture.detectChanges();
+        expect(examDetailComponent).to.be.ok;
+        // stand in for other properties too who are simply loaded from the exam and displayed in spans
+        const titleSpan = examDetailComponentFixture.debugElement.query(By.css('#examTitle')).nativeElement;
+        expect(titleSpan).to.be.ok;
+        expect(titleSpan.innerHTML).to.equal(exam.title);
+    });
+
+    it('should correctly route to edit subpage', fakeAsync(() => {
+        const location = examDetailComponentFixture.debugElement.injector.get(Location);
+        examDetailComponentFixture.detectChanges();
+        const editButton = examDetailComponentFixture.debugElement.query(By.css('#editButton')).nativeElement;
+        editButton.click();
+        examDetailComponentFixture.whenStable().then(() => {
+            expect(location.path()).to.equal('/course-management/1/exams/1/edit');
+        });
+    }));
+
+    it('should correctly route to student exams subpage', fakeAsync(() => {
+        const location = examDetailComponentFixture.debugElement.injector.get(Location);
+        examDetailComponentFixture.detectChanges();
+        const editButton = examDetailComponentFixture.debugElement.query(By.css('#studentExamsButton')).nativeElement;
```

Comment:
Fwiw, this is not the edit button. :P

Reply:
Done in : 3d8e8b2187c45a0a6a94842e37a220b70ee45ca2